src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/DisableRuntimeMarshalling.cs

```diff
@@ -169,22 +241,22 @@ public void AnalyzeMethod(SymbolAnalysisContext context)
 
                 if (dllImportData.SetLastError)
                 {
-                    context.ReportDiagnostic(method.CreateDiagnostic(FeatureUnsupportedWhenRuntimeMarshallingDisabled, SetLastErrorTrue));
+                    context.ReportDiagnostic(method.CreateDiagnostic(FeatureUnsupportedWhenRuntimeMarshallingDisabledSetLastErrorTrue));
                 }
 
                 if (!method.MethodImplementationFlags().HasFlag(System.Reflection.MethodImplAttributes.PreserveSig))
                 {
-                    context.ReportDiagnostic(method.CreateDiagnostic(FeatureUnsupportedWhenRuntimeMarshallingDisabled, HResultSwapping));
+                    context.ReportDiagnostic(method.CreateDiagnostic(FeatureUnsupportedWhenRuntimeMarshallingDisabledHResultSwapping));
                 }
 
-                if (_lcidConversionAttribute is not null && method.GetAttributes(_lcidConversionAttribute).Any())
+                if (_lcidConversionAttribute is not null && method.HasAttribute(_lcidConversionAttribute))
```

Comment:
`_lcidConversionAttribute is not null` is redundant, `HasAttribute` performs the null check.